src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -84,13 +89,49 @@ private static double computeTotalDemand(Component component, double lossCoeffic
         return totalLoad * (1. + lossCoefficient / 100.);
     }
 
-    private double computeTotalAmountFixedSupply(int numCC) {
+    private double computeTotalAmountFixedSupply(Component component) {
         double totalAmountFixedSupply = 0.;
-        totalAmountFixedSupply += fixedSupplyGenerators.get(numCC).stream().filter(generator -> generator.getTerminal().isConnected())
+        totalAmountFixedSupply += fixedSupplyGenerators.get(component.getNum()).stream().filter(generator -> generator.getTerminal().isConnected())
             .mapToDouble(Generator::getTargetP).sum();
         return totalAmountFixedSupply;
     }
 
+    private double computeHvdcBalance(Component component) {
+        AtomicDouble balance = new AtomicDouble(0.);
+
+        component.getBusStream().forEach(bus -> {
+            double hdvcFlow = Stream.concat(bus.getLccConverterStationStream(), bus.getVscConverterStationStream())
+                .filter(station -> {
+                    // Keep only hvdc linking to another synchronous component
+                    HvdcLine hvdcLine = station.getHvdcLine();
+                    HvdcConverterStation<?> station1 = hvdcLine.getConverterStation1();
+                    HvdcConverterStation<?> station2 = hvdcLine.getConverterStation2();
+                    boolean station2NotInComponent = station1.getId().equals(station.getId()) &&
```

Comment:
Would be nice to have a static function that extracts the component from a converter station.

Reply:
Done